From 65c9fecec9e67eb88980cb414e182fdbf7db2511 Mon Sep 17 00:00:00 2001
From: rramasamy <rramasamy@maxlinear.com>
Date: Wed, 18 Jun 2025 11:28:28 +0800
Subject: [PATCH] agent: son_slave_thread: associated sta link metrics
 correction

create Associated Sta Extended Link Metrics TLV after filling the Associated
STA Link Metrics TLV which fixes adding the garbage value in Link Metrics TLV.

Closes PPM-3362

Signed-off-by: rramasamy <rramasamy@maxlinear.com>
---
 agent/src/beerocks/slave/son_slave_thread.cpp | 30 +++++++++----------
 1 file changed, 15 insertions(+), 15 deletions(-)

diff --git a/agent/src/beerocks/slave/son_slave_thread.cpp b/agent/src/beerocks/slave/son_slave_thread.cpp
index e873ede21a..632d815371 100644
--- a/agent/src/beerocks/slave/son_slave_thread.cpp
+++ b/agent/src/beerocks/slave/son_slave_thread.cpp
@@ -3596,21 +3596,6 @@ bool slave_thread::handle_cmdu_monitor_message(const std::string &fronthaul_ifac
             return false;
         }
 
-        // adding (currently empty) an associated sta EXTENDED link metrics tlv.
-        // The values will be filled part of PPM-1259
-        auto extended = cmdu_tx.addClass<wfa_map::tlvAssociatedStaExtendedLinkMetrics>();
-        if (!extended) {
-            LOG(ERROR) << "adding wfa_map::tlvAssociatedStaExtendedLinkMetrics failed";
-            return false;
-        }
-
-        extended->associated_sta() = response_in->sta_mac();
-
-        if (!extended->alloc_metrics_list(response_in->bssid_info_list_length())) {
-            LOG(ERROR) << "allocation of per BSS STA metrics failed";
-            return false;
-        }
-
         auto db = AgentDB::get();
 
         for (size_t i = 0; i < response_out->bssid_info_list_length(); ++i) {
@@ -3640,6 +3625,21 @@ bool slave_thread::handle_cmdu_monitor_message(const std::string &fronthaul_ifac
             bss_out.sta_measured_uplink_rcpi_dbm_enc = bss_in.sta_measured_uplink_rcpi_dbm_enc;
         }
 
+        // adding (currently empty) an associated sta EXTENDED link metrics tlv.
+        // The values will be filled part of PPM-1259
+        auto extended = cmdu_tx.addClass<wfa_map::tlvAssociatedStaExtendedLinkMetrics>();
+        if (!extended) {
+            LOG(ERROR) << "adding wfa_map::tlvAssociatedStaExtendedLinkMetrics failed";
+            return false;
+        }
+
+        extended->associated_sta() = response_in->sta_mac();
+
+        if (!extended->alloc_metrics_list(response_in->bssid_info_list_length())) {
+            LOG(ERROR) << "allocation of per BSS STA metrics failed";
+            return false;
+        }
+
         LOG(DEBUG) << "Send AssociatedStaLinkMetrics to controller, mid = " << mid;
         send_cmdu_to_controller({}, cmdu_tx);
         break;
-- 
GitLab

